app/src/main/java/de/triology/cas/services/RegistryEtcd.java

```diff
@@ -44,48 +44,51 @@ public RegistryEtcd(EtcdClient etcd) {
     }
 
     @Override
-    public List<String> getDogus() {
+    public List<String> getInstalledDogusWhichAreUsingCAS() {
         log.debug("Get Dogus from registry");
         try {
-            List<EtcdKeysResponse.EtcdNode> nodes = etcd.getDir("/dogu").recursive().send().get().getNode().getNodes();
-            return convertNodesToStringList(nodes);
+            List<EtcdKeysResponse.EtcdNode> nodes = etcd.getDir(DOGU_DIR).send().get().getNode().getNodes();
+            return extractDogusFromDoguRootDir(nodes);
         } catch (IOException | EtcdException | EtcdAuthenticationException | TimeoutException e) {
-            log.error("Failed to getDogus: ", e);
+            log.error("Failed to getInstalledDogusWhichAreUsingCAS: ", e);
             throw new RegistryException(e);
         }
     }
 
     @Override
     public String getFqdn() {
-        log.debug("Get FQDN from registry");
+        return getEtcdValueForKey("/config/_global/fqdn");
+    }
+
+    public String getEtcdValueForKey(String key) {
+        log.debug("Get " + key + " from registry");
         try {
-            return etcd.get("/config/_global/fqdn").send().get().getNode().getValue();
+            return etcd.get(key).send().get().getNode().getValue();
         } catch (IOException | EtcdException | EtcdAuthenticationException | TimeoutException e) {
-            log.error("Failed to getFqdn: ", e);
+            log.warn("Failed to getEtcdValueForKey: ", e);
             throw new RegistryException(e);
         }
     }
 
     public URI getCasLogoutUri(String doguname) throws GetCasLogoutUriException {
         JSONObject doguMetaData;
         try {
-            EtcdKeysResponse.EtcdNode node = getDoguNodeFromEtcd(doguname);
-            doguMetaData = getCurrentDoguNode(node);
+            doguMetaData = getCurrentDoguNode(doguname);
             JSONObject properties;
             if (doguMetaData != null) {
                 properties = getPropertiesFromMetaData(doguMetaData);
             } else {
                 throw new GetCasLogoutUriException("Could not get dogu metadata");
             }
             return getLogoutUriFromProperties(properties);
-        } catch (ClassCastException | NullPointerException | ParseException | URISyntaxException | GetDoguNodeFromEtcdException e) {
+        } catch (ClassCastException | NullPointerException | ParseException | URISyntaxException | RegistryException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1166)